br/pkg/storage/azblob.go

```diff
@@ -0,0 +1,510 @@
+		cred, err := azidentity.NewClientSecretCredential(tenantId, clientId, clientSecret, nil)
+		if err != nil {
+			log.Warn("Failed to get azure token credential but environment variables exist, try to use shared key.", zap.String("tenantId", tenantId), zap.String("clientId", clientId), zap.String("clientSecret", "?"))
+		} else {
+			// send account-name to TiKV
+			if opts != nil && opts.SendCredentials {
+				options.AccountName = accountName
+			}
+			return &tokenClientBuilder{
+				cred,
+				accountName,
+				serviceURL,
+			}, nil
+		}
+	}
+
+	var sharedKey string
+	if val := os.Getenv("AZURE_STORAGE_KEY"); len(val) > 0 {
+		log.Info("Get azure sharedKey from environment variable $AZURE_STORAGE_KEY")
+		sharedKey = val
+	} else {
+		return nil, errors.New("cannot find any credential info to access azure blob storage")
+	}
+
+	cred, err := azblob.NewSharedKeyCredential(accountName, sharedKey)
+	if err != nil {
+		return nil, errors.Annotate(err, "Failed to get azure sharedKey credential")
+	}
+	// if BR can only get credential info from environment variable `sharedKey`,
+	// BR will send it to TiKV so that there is no need to set environment variable for TiKV.
+	if opts != nil && opts.SendCredentials {
+		options.AccountName = accountName
+		options.SharedKey = sharedKey
+	}
+	return &sharedKeyClientBuilder{
+		cred,
+		accountName,
+		serviceURL,
+	}, nil
+}
+
+type AzureBlobStorage struct {
+	options *backuppb.AzureBlobStorage
+
+	containerClient azblob.ContainerClient
+
+	accessTier azblob.AccessTier
+}
+
+func newAzureBlobStorage(ctx context.Context, options *backuppb.AzureBlobStorage, opts *ExternalStorageOptions) (*AzureBlobStorage, error) {
+	clientBuilder, err := getAzureServiceClientBuilder(options, opts)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return newAzureBlobStorageWithClientBuilder(ctx, options, clientBuilder)
+}
+
+func newAzureBlobStorageWithClientBuilder(ctx context.Context, options *backuppb.AzureBlobStorage, clientBuilder ClientBuilder) (*AzureBlobStorage, error) {
+
+	serviceClient, err := clientBuilder.GetServiceClient()
+	if err != nil {
+		return nil, errors.Annotate(err, "Failed to create azure service client")
+	}
+
+	respIter := serviceClient.ListContainers(nil)
+	containerExist := false
+RESPITER:
+	for {
+		err := respIter.Err()
+		if err != nil {
+			return nil, errors.Annotate(err, "Failed to list azure containers")
+		}
+
+		if !respIter.NextPage(ctx) {
+			err := respIter.Err()
+			if err != nil {
+				return nil, errors.Annotate(err, "Failed to list azure containers")
+			}
+			break RESPITER
+		}
+
+		for _, container := range respIter.PageResponse().ContainerItems {
+			if *container.Name == options.Bucket {
+				containerExist = true
+				break RESPITER
+			}
+		}
+	}
+
+	respIter.PageResponse().RawResponse.Body.Close()
+	containerClient := serviceClient.NewContainerClient(options.Bucket)
+	if !containerExist {
+		_, err := containerClient.Create(ctx, nil)
```

Comment:
according to document. 
```
Create creates a new container within a storage account. If a container with the same name already exists, the operation fails. For more information, see https://docs.microsoft.com/rest/api/storageservices/create-container.
```

so may be we can try create before, if failed then ignore create error?